### ISSUE TITLE
Support for extracting/bundling headers in C/C++

### ISSUE THREAD

**Issue by przydatek** [bazelbuild/bazel#5192]
### Description of the problem / feature request:

Getting C/C++ header files for a library built via Bazel.

### Feature requests: what underlying problem are you trying to solve with this feature?

When building C/C++ library to be used in a project without Bazel one needs also a set of the header files with the API that the library offers.  It would be nice if Bazel could add some support for creation of  a bundle of the relevant headers, e.g. as a filegroup.

### What operating system are you running Bazel on?

Linux

### What's the output of `bazel info release`?

release 0.13.0

### What's the output of `git remote get-url origin ; git rev-parse master ; git rev-parse HEAD` ?

https://github.com/google/tink
eb6673aff24a09947450d8922702f5f43fa9aaa2
eb6673aff24a09947450d8922702f5f43fa9aaa2

###  Have you found anything relevant by searching the web?

It seems the only way to do it now is "manually", see e.g. TensorFlow: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/lib_package/BUILD#L57
and https://github.com/tensorflow/tensorflow/issues/3536

**Comment by mhlopko** [comment]
This will be easily solved in Skylark when #4570 is implemented, I propose to wait until that happens.

**Comment by kingstarcraft** [comment]
If you build windows with bazel, you can try [my tools](https://github.com/kingstarcraft/Tensorflow-C-Install.git) to export the header. I have test my tool can work with tensorflow v1.12.0.

**Comment by oquenchil** [comment]
You can use the CcInfo provider for this:
https://docs.bazel.build/versions/master/skylark/lib/CcInfo.html

**Comment by gccunha1** [comment]
> You can use the CcInfo provider for this: https://docs.bazel.build/versions/master/skylark/lib/CcInfo.html

Can someone provide a simple example on how to do this? I've been looking at the documentation and only found this example: https://github.com/bazelbuild/rules_cc/blob/main/examples/my_c_archive/my_c_archive.bzl 

For me it's a bit hard to follow as I haven't really been using bazel at its full capacity
